tests/unittests/sources/test_ec2.py

```diff
@@ -582,42 +590,45 @@ def test_network_config_cached_property_refreshed_on_upgrade(self, m_dhcp):
             platform_data=self.valid_platform_data,
             sys_cfg={"datasource": {"Ec2": {"strict_id": True}}},
             md={"md": old_metadata},
+            mocker=mocker,
+            tmpdir=tmpdir,
         )
-        self.assertTrue(ds.get_data())
+        assert True is ds.get_data()
 
-        # Workaround https://github.com/getsentry/responses/issues/212
-        if hasattr(self.responses, "_urls"):
-            # Can be removed when Bionic is EOL
-            for index, url in enumerate(self.responses._urls):
+        def _remove_md(resp_list: List) -> None:
+            for index, url in enumerate(resp_list):
                 if url["url"].startswith(
                     "http://169.254.169.254/2009-04-04/meta-data/"
                 ):
-                    del self.responses._urls[index]
-        elif hasattr(self.responses, "_matches"):
+                    del resp_list[index]
+
+        # Workaround https://github.com/getsentry/responses/issues/212
+        if hasattr(responses, "_urls"):
+            # Can be removed when Bionic is EOL
+            _remove_md(responses._urls)
+        elif hasattr(responses, "_default_mock") and hasattr(
+            responses._default_mock, "_urls"
+        ):
+            # Can be removed when Bionic is EOL
+            _remove_md(responses._default_mock._urls)
+        elif hasattr(responses, "_matches"):
             # Can be removed when Focal is EOL
-            for index, response in enumerate(self.responses._matches):
-                if response.url.startswith(
-                    "http://169.254.169.254/2009-04-04/meta-data/"
-                ):
-                    del self.responses._matches[index]
+            _remove_md(responses._matches)
```

Comment:
I'm not seeing this being any issue running tests on either Bionic or Focal if we remove this section. I'm not sure it's 
on focal now tox is pulling down:
```
responses==0.25.0
requests==2.31.0
```
These versions don'tt seem to represent this duplicated/incorrect stale registration of responses. 
Since we no longer run CI on Bionic, I'm even less concerned about dropping the Bionic _remove_mds as well.

Reply:
We still have [lowest-supported](https://github.com/canonical/cloud-init/blob/2f9812e805f8e66feaf2689384ea6d669305d9a5/.github/workflows/unit.yml#L23) unit test target which runs on `py36` and requires this block.

Are you suggesting we should move `lowest-supported` to be run in focal and bump the dependencies in `tox.ini`?